### PR TITLE
Adds CursorStyles option for terminal

### PR DESCRIPTION
**Pull request by czuria1**
## Summary

Luckily in version [1.2.0](https://github.com/migueldeicaza/SwiftTerm/commit/131a52bf3b1e9e8f8ed8d8298d3112972eb9ce22) of `SwiftTerm`, they’ve introduced `CursorStyle` which provides multiple styles for the cursor. In issue https://github.com/AuroraEditor/AuroraEditor/issues/482, it was mentioned that currently the cursor blocks visibility of a character when going over it. Therefore, by introducing cursor styles, the user can now choose their preference of shape as well as whether they'd like it to blink or not allowing them to see the character behind the cursor.

## Screenshots/GIFs

CleanShot 2023-10-16 at 23 12 51

CleanShot 2023-10-16 at 23 10 17@2x

## Checklist

Please ensure all of the following are completed before submitting the PR:

- [x] Code has been tested and verified.
- [x] Documentation has been updated to reflect the changes.
- [x] All tests pass successfully.
- [x] Code follows the project's coding guidelines and style.
- [x] The branch is up-to-date with the latest changes from the main branch.
- [ ] Reviewed by at least one other contributor.

## Related Issues

This PR addresses the following issue(s): https://github.com/AuroraEditor/AuroraEditor/issues/482

### CODE REVIEW

File: AuroraEditor/Base/AppPreferences/UI/TerminalPreferences/TerminalPreferencesView.swift [hunk 1]
```diff
@@ -79,6 +83,26 @@ public struct TerminalPreferencesView: View {
             }
         }
     }
+
+    private var cursorSelector: some View {
```
Comment (czuria1) [inline AuroraEditor/Base/AppPreferences/UI/TerminalPreferences/TerminalPreferencesView.swift:87]: Referenced the Terminal for this portion:
image

**Comment by nanashili** [comment]
Looks good to me!

**Review by nanashili — APPROVED** [review]

**Comment by 0xWDG** [comment]
@all-contributors please add @czuria1 for bug.

**Comment by 0xWDG** [comment]
Thanks for your contribution!

**Pull request merged**